sys/riscv/include/vmparam.h

```diff
@@ -160,10 +160,32 @@
 #define	PHYS_IN_DMAP(pa)	((pa) >= DMAP_MIN_PHYSADDR && \
     (pa) < DMAP_MAX_PHYSADDR)
 /* True if va is in the dmap range */
+#ifdef __CHERI_PURE_CAPABILITY__
+#define	VIRT_IN_DMAP(va)						\
+	cheri_is_address_inbounds(dmap_capability, (va))
+#else
 #define	VIRT_IN_DMAP(va)	((va) >= DMAP_MIN_ADDRESS && \
     (va) < (dmap_max_addr))
+#endif
```

Comment:
I'm not a fan of reusing capability bounds as a software sanity check. If you forget to set the cap bounds then you lose your software bounds check too, whereas if they're independent then you have redundancy and the cap bounds are there as a fallback in case you ever forget the software check. It also reduces the diff to upstream.

Reply:
I did this on the assumption that it was cheaper. :)  That said, I can't use the upstream check as is as I'll have to add casts to get the ptraddr_t.